### ISSUE TITLE
Can't use dbt-sqlserver as part of poetry managed project due to incompatible version declaration

### ISSUE THREAD

**Issue by drobert-bfm** [dbt-msft/dbt-sqlserver#462]
dbt-sqlserver declares itself to be incompatible with versions of dbt-core greater than 1.5.0, whereas dbt-core is now at 1.7+. The problem is in setup.py, under the following:

install_requires=[
        "dbt-core~=1.4.5",
        "pyodbc~=4.0.35,!=4.0.36,!=4.0.37",
        "azure-identity>=1.12.0",
    ],

Please bring compatibility up to current dbt-core version.

**Comment by mhurenkamp-groveko** [comment]
You may not have yet seen the following related issue: https://github.com/dbt-msft/dbt-sqlserver/issues/441 , which gives some insight on the future support of this adapter.

To my knowledge, no news has arrived from Microsoft to officially support a dbt adapter for MS SQL, but it will support dbt on Fabric. For the SQL Server version we can 'borrow' code from the Fabric adapter.

@cody-scott was kind enough to share a pull request https://github.com/dbt-msft/dbt-sqlserver/pull/461 to bring the adapter up to 1.7.2 using the Fabric adapter. I suggest following the status of this PR.

**Comment by schlich** [comment]
@drobert-bfm 

If this is still relevant, can you test your project using the pre-release version `1.7.2rc2` and let us know if this resolves your issues?

**Comment by drobert-bfm** [comment]
Yes, it resolves the initial tests I've been running. It's not a very deep test, but basic materializations seem to work.
